src/libostree/ostree-mutable-tree.c

```diff
@@ -47,10 +47,20 @@ struct OstreeMutableTree
 {
   GObject parent_instance;
 
+  /* This is the checksum of the Dirtree object that corresponds to the current
+   * contents of this directory.  contents_checksum can be NULL if the SHA was
+   * never calculated or contents of the mtree has been modified.  Even if
+   * contents_checksum is not NULL it may be out of date. */
```

Comment:
> Even if contents_checksum is not NULL it may be out of date.

Is that true though? We invalidate it before it gets out of date, no?

Reply:
No, see the comment in `ostree_mutable_tree_get_contents_checksum`:

https://github.com/ostreedev/ostree/blob/e120a6b1198aaa785533c76316898f15a804dae1/src/libostree/ostree-mutable-tree.c#L120-L135

We do invalidate it if this mtree is modified, but not if subdirectories of this mtree are modified.

In `state == LAZY` it is always accurate.